### ISSUE TITLE
failed decryption creates empty output file

### ISSUE THREAD

**Issue by barak** [Tarsnap/scrypt#120]
This is with version 1.2.1. A failed decryption creates an empty output file.
```
$ scrypt dec /dev/null foo-dec
Please enter passphrase: 
scrypt: Input is not valid scrypt-encrypted block
$ ls -l foo*
-rw-rw-r-- 1 barak barak 0 Feb 23 17:12 foo-dec
```
This also happens when decrypting a valid file with the wrong passphrase.

**Comment by gperciva** [comment]
Hi @barak, I've addressed this in #121.

**Comment by barak** [comment]
Cool, thanks @gperciva.

(Unless you instruct me otherwise, I'll wait until you make a stable release before putting this in the Debian package.)

**Comment by gperciva** [comment]
Fixed in #144.
